### PR TITLE
steps: add codechecks and timemory variants

### PR DESCRIPTION
**Pull request by tristan0x**

### CODE REVIEW

File: var/spack/repos/builtin/packages/steps/package.py [hunk 1]
```diff
@@ -42,13 +46,16 @@ class Steps(CMakePackage):
     depends_on("py-scipy", type=("build", "test"))
     depends_on("py-matplotlib", type=("build", "test"))
     depends_on("python")
-
+    depends_on('git', type='build', when='+codechecks')
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/steps/package.py]: IIRC, git is prerequisite for spack. So no need to add it here.

Comment (tristan0x) [reply var/spack/repos/builtin/packages/steps/package.py]: dependency removed in fda07d7

File: var/spack/repos/builtin/packages/steps/package.py [hunk 2]
```diff
@@ -42,13 +46,15 @@ class Steps(CMakePackage):
     depends_on("py-scipy", type=("build", "test"))
     depends_on("py-matplotlib", type=("build", "test"))
     depends_on("python")
-
+    depends_on("py-cmake-format", type="build", when="+codechecks")
+    depends_on('py-pre-commit', type='build', when='+codechecks')
+    depends_on('py-pyyaml', type='build', when='+codechecks')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/steps/package.py:51]: Stupid question. Shouldn't you enable something explicitly when using `+codechecks`? Maybe I just need more :coffee: …

Comment (tristan0x) [reply var/spack/repos/builtin/packages/steps/package.py:51]: I do not understand the question, sorry. Maybe I just need more ☕

Comment (matz-e) [reply var/spack/repos/builtin/packages/steps/package.py:51]: You added a variant and dependencies, but no additional cmake flags?

Comment (matz-e) [reply var/spack/repos/builtin/packages/steps/package.py:51]: I.e., in the code below, I'd expect a

```
if "+codechecks" in spec:
```

Comment (tristan0x) [reply var/spack/repos/builtin/packages/steps/package.py:51]: > I.e., in the code below, I'd expect a
> 
> ```
> if "+codechecks" in spec:
> ```

Good point, but that will come later if that's ok for you.

**Review by matz-e — APPROVED** [review]
I'll trust you to add the missing bits discussed.

**Pull request merged**